src/App/DataSource/Email/Email.php

```diff
@@ -219,11 +179,25 @@ public function fetch($limit = false)
                 return [];
             }
 
-            $last_uid = $this->messageRepo->getLastUID('email');
-            $max_range = $last_uid + $limit;
-            $search_string = $last_uid ? $last_uid + 1 . ':' . $max_range : '1:' . $max_range;
+            $mailboxinfo = imap_check($connection);
+
+            Log::info("Connected to $inbox", [$username, $password, $mailboxinfo]);
 
-            $emails = imap_fetch_overview($connection, $search_string, FT_UID);
+            $last_uid = $this->messageRepo->getLastUID('email');
+            if ($last_uid > 0) {
```

Comment:
This might still fail if you change you email settings, because it won't reset the last uid. We could update it to save that to config and reset similar to the twitter provider.

Reply:
that'd be cool. This could happen with admins switching the admin email most likely, I think.